app/gui/src/dashboard/layouts/AssetContextMenu.tsx

```diff
@@ -248,19 +251,31 @@ export default function AssetContextMenu(props: AssetContextMenuProps) {
               }}
             />
           )}
-        {asset.type === backendModule.AssetType.project && isCloud && (
+        {asset.type === backendModule.AssetType.project && isCloud && enableHybridExecution && (
           <ContextMenuEntry
             hidden={hidden}
             action="run"
             isDisabled={!canOpenProjects}
             tooltip={disabledTooltip}
-            doAction={() => {
-              openProjectMutation.mutate({
-                id: asset.id,
-                title: asset.title,
-                parentId: asset.parentId,
-                type: state.backend.type,
-                inBackground: true,
+            doAction={async () => {
+              invariant(localBackend != null, 'Local Backend is null')
+              const parentId = await remoteBackend.downloadProject(asset.id)
```

Comment:
I think the `downloadProject` should return all the needed data, shouldn't it?
Adding extra `fetch` looks unnecessary

Reply:
The downloaded cloud project does not have `.enso/project.json` metadata and we cannot run it locally. We need to execute `listDirectory` to recreate missing metadata and get the project id.